Update and add support to Keychron Q1v1

adophoxia opened the pull request:
Add support for ANSI Knob and ISO variants of the Q1v1 while also updating the Q1v1 no-knob ANSI variant.

adophoxia committed:
Port VIAL to Keychron Q1v1

Add support for Q1v1 ANSI Knob and ISO variants
- Add `LTO_ENABLE = yes` rule to `rules.mk` in keyboard folders to reduce firmware size
- Add `ENCODER_MAP_ENABLE ` rule for Q1v1 ANSI knob variant in keymap level to leverage support for rotary encoder

adophoxia committed:
Update Keychron Q1v1

Add `LTO_ENABLE = yes` rule to `rules.mk` in keyboard folder to reduce firmware size

xyzz commented on keyboards/keychron/q1/rev_0100/rules.mk:
```diff
@@ -19,3 +19,4 @@ AUDIO_ENABLE = no           # Audio output
 DIP_SWITCH_ENABLE = yes
 RGB_MATRIX_ENABLE = yes
 RGB_MATRIX_DRIVER = IS31FL3733
+LTO_ENABLE = yes
```
Any changes should go under the `vial` keymap

adophoxia replied:
Made the necessary changes for LTO_ENABLE rule relocation.

xyzz commented on keyboards/keychron/q1/rev_0101/rules.mk:
```diff
@@ -19,4 +19,5 @@ AUDIO_ENABLE = no           # Audio output
-RGB_MATRIX_DRIVER = IS31FL3733
\ No newline at end of file
+RGB_MATRIX_DRIVER = IS31FL3733
+LTO_ENABLE = yes
```
as above

adophoxia replied:
As above

xyzz commented on keyboards/keychron/q1/rev_0102/rules.mk:
```diff
@@ -18,4 +18,5 @@ RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
-RGB_MATRIX_DRIVER = IS31FL3733
\ No newline at end of file
+RGB_MATRIX_DRIVER = IS31FL3733
+LTO_ENABLE = yes
```
as above

adophoxia replied:
As above

adophoxia committed:
Update rules.mk

Moved `LTO_ENABLE = yes` rule from keyboard level to keymap level in the `rules.mk`

xyzz commented on keyboards/keychron/q1/rev_0100/rules.mk:
```diff
@@ -18,4 +18,4 @@ RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
 DIP_SWITCH_ENABLE = yes
 RGB_MATRIX_ENABLE = yes
-RGB_MATRIX_DRIVER = IS31FL3733
+RGB_MATRIX_DRIVER = IS31FL3733
```
there are still changes in this file

adophoxia replied:
The only changes made was removing the "LTO_ENABLE" rule and moving it to the the keymap "rules.mk". I did check again to make sure that nothing else was affected besides that one line, to which I found nothing else changed.

adophoxia committed:
Merge branch 'vial-kb:vial' into vial-keychron-q1

adophoxia commented:
Wanting to see if the changes requested in `rules.mk` is still needed.
